src/map.ts

```diff
@@ -0,0 +1,244 @@
+    longitude: 2.1610199,
+    zoom: 13,
+    bearing: 10,
+    pitch: 45,
+    ...CITY_STATE_TRANSITIONS
+  }
+
+};
+
+let deck : Deck;
+
+// Render the right layers and initial view state each time a story is selected
+
+let selectedStory = "dynamicTiling";
+
+export function selectStoryMap(story: string) {
+  selectedStory = story;
+  deck.setProps({
+    initialViewState: INITIAL_VIEW_STATE[story]
+  });
+  render();
+}
+
+// Get the category to highlight
+
+let selectedOsmCategory = "Financial";
+
+export function selectOsmCategory(category: string) {
+  selectedOsmCategory = category;
+  render();
+}
+
+// Filter the query based on the selected year
+
+let selectedYear = 2010;
+
+export function selectYear(year: number) {
+  selectedYear = year;
+  console.log(selectedYear);
+  render();
+}
+
+// A longer query to aggregate points (accidents) in polygons (states)
+
+const countAccidentsPerState = `
+
+SELECT
+    polygonsb.*,
+    COALESCE(__count.count, 0) AS count
+  FROM \`carto-demo-data.demo_tables.usa_states_boundaries\` polygonsb
+  LEFT JOIN
+    (
+      SELECT
+        polygons.code_hasc poligonid,
+        count(*) count
+      FROM
+        \`carto-demo-data.demo_tables.usa_states_boundaries\` as polygons,
+        \`carto-demo-data.demo_tables.riskanalysis_railroad_accidents\` as points
+      WHERE
+          ST_INTERSECTS(
+            polygons.geom,
+            points.geom
+          ) AND
+          points.year = @selectedYear
+      GROUP BY
+          polygons.code_hasc
+    ) __count
+  ON __count.poligonid = polygonsb.code_hasc;
+
+`
+
+// For the tileset story, move the camera to a 3d view and update it based on the buttons
+
+export function moveToCity(city: string) {
+  const cityLocation: object = getLocation(city);
+
+  const CITY_VIEW_STATE = {
+    latitude: cityLocation.latitude,
+    longitude: cityLocation.longitude,
+    zoom: 13,
+    bearing: 10,
+    pitch: 45,
+    ...CITY_STATE_TRANSITIONS
+  };
+
+  deck.setProps({
+    initialViewState: CITY_VIEW_STATE
+  });
+  render();
+}
+
+
+// Our render function: it sets the deck.gl layers based on our inputs
+
+export function render() {
+
+  deck.setProps({
+    layers: [
+      new CartoLayer({
+        id: 'pois',
+        connection: 'carto_dw',
+        type: MAP_TYPES.QUERY,
+        data: 'SELECT * FROM carto-demo-data.demo_tables.osm_pois_usa',
+        opacity: 1,
+        getFillColor: colorCategories({
+          attr: 'group_name',
+          domain: [selectedOsmCategory],
+          colors: [[3, 111, 226]],
+          othersColor: [3, 111, 226, 100]
+        }),
+        getLineColor: colorCategories({
+          attr: 'group_name',
+          domain: [selectedOsmCategory],
+          colors: [[255, 255, 255]],
+          othersColor: [3, 111, 226, 0]
+        }),
+        getPointRadius: 50,
+        getLineWidth: 10,
+        pointRadiusMinPixels: 1,
+        lineWidthMinPixels: 0.3,
+        visible: selectedStory === "dynamicTiling",
```

Comment:
In general you're using single quote, I'd move to single everything

Reply:
✅